scarb/src/bin/scarb/args.rs

```diff
@@ -132,12 +131,25 @@ pub enum Command {
     Metadata(MetadataArgs),
     /// Create a new Scarb package at <PATH>.
     New(NewArgs),
+    /// Run arbitrary package scripts.
+    Run(ScriptsRunnerArgs),
 
     /// External command (`scarb-*` executable).
     #[command(external_subcommand)]
     External(Vec<OsString>),
 }
 
+/// Arguments accepted by the `run` command.
+#[derive(Parser, Clone, Debug)]
+#[clap(trailing_var_arg = true)]
+pub struct ScriptsRunnerArgs {
+    pub script: Option<SmolStr>,
+    #[command(flatten)]
+    pub packages_filter: PackagesFilter,
```

Comment:
please ensure that `scarb run script -p package` goes to package filter, but `scarb run script -- -p package` goes as an argument to this script